### PR TITLE
Databricks Splunk Integration v1.2.0 Code Changes

### PR DESCRIPTION
**Pull request by fatemag-crest**
Updated the Add-on to allow non-admin users to execute the custom commands.
Added support for mutiple account creation.
Updated the user-agent to include logged in user.
Updated Unit test cases.

### CODE REVIEW

**Comment by alexott** [comment]
@fatemag-crest please add more tests, as this PR won't be merged without meeting the quality targets

**Comment by dhruvilbhatt-crest** [comment]
@alexott - We had this discussed with Zafer, and he confirmed we are good with the current coverage which is 81.05%. The required coverage is 80% as mentioned [here](https://github.com/databrickslabs/splunk-integration/blob/master/tests/.coveragerc#L19). 

Can you check why GitHub is failing this even if the coverage is above the mentioned limit?

File: app/README.md [hunk 1]
```diff
@@ -48,46 +53,53 @@ The Databricks Add-on for Splunk is used to query Databricks data and execute Da
...
+| Databricks Cluster Name   | Name of the Databricks cluster to use for query and notebook execution. A user can override this value while executing the custom command.                                                                                                         |      No
+| Authentication Method     | SingleSelect: Authentication via Azure Active Directory or using a Personal Access Token |      Yes
+| Databricks Access Token   | [Auth: Personal Access Token] Databricks personal access token to use for authentication. Refer [Generate Databricks Access Token](https://docs.databricks.com/dev-tools/api/latest/authentication.html#generate-a-personal-access-token) document to generate the access token. |      Yes                                                                                              |
+| Client Id   | [Auth: Azure Active Directory] Azure Active Directory Client Id from your Azure portal.|      Yes
+| Tenant Id   | [Auth: Azure Active Directory] Databricks Application(Tenant) Id from your Azure portal.|      Yes
+| Client Secret  | [Auth: Azure Active Directory] Azure Active Directory Client Secret from your Azure portal.|      Yes
```
Comment (alexott) [inline app/README.md:86]: We may need to rename them to be explicit for Azure as we're getting OAuth support for AWS as well.

File: app/appserver/static/img/add_databricks_admin_user.webm
Comment (alexott) [inline app/appserver/static/img/add_databricks_admin_user.webm:1]: Why not .png ?

File: app/bin/databricks_com.py [hunk 2]
```diff
@@ -8,34 +8,65 @@
...
-            raise Exception("Addon is not configured. Navigate to addon's configuration page to configure the addon.")
+            raise Exception(
+                "Addon is not configured. Navigate to addon's configuration page to configure the addon."
+            )
         self.databricks_instance_url = "{}{}".format("https://", databricks_instance.strip("/"))
         self.request_headers = {
             "Authorization": "Bearer {}".format(self.databricks_token),
             "Content-Type": "application/json",
-            "User-Agent": const.USER_AGENT_CONST
+            "User-Agent": "{}".format(const.USER_AGENT_CONST),
```
Comment (alexott) [inline app/bin/databricks_com.py:65]: what is the point of using `format` here?

File: app/bin/databricks_common_utils.py [hunk 3]
```diff
@@ -255,66 +259,151 @@ def format_to_json_parameters(params):
...
-            elif resp.status_code == 429:
-                msg = "API limit exceeded. Please try again after some time.",
-            elif resp.status_code not in (200, 201):
-                msg = 'Response status: {}. Unable to validate Azure Active Directory Credentials.'\
-                    'Check logs for more details.'.format(str(resp.status_code))
+            if "resp" in locals():
+                error_code = resp.json().get("error_codes")
+                if error_code:
+                    error_code = str(error_code[0])
+                if error_code in list(const.ERROR_CODE.keys()):
```
Comment (alexott) [inline app/bin/databricks_common_utils.py:385]: if `const.ERROR_CODE` is a map, then it's just enough: `if error_code in const.ERROR_CODE`

**Review by alexott — APPROVED** [review]
the comments needs to be addressed in the follow up PRs

**Pull request merged**